django_app/redbox_app/redbox_core/models.py

```diff
@@ -55,22 +55,37 @@ class ProcessingStatusEnum(models.TextChoices):
 
 
 class File(UUIDPrimaryKeyBase, TimeStampedModel):
-    processing_status = models.CharField(choices=ProcessingStatusEnum.choices, null=False, blank=False)
-    original_file = models.FileField(storage=settings.BUCKET_NAME)
+    processing_status = models.CharField(
+        choices=ProcessingStatusEnum.choices, null=False, blank=False
+    )
+    original_file = models.FileField(storage=settings.DEFAULT_FILE_STORAGE)
     user = models.ForeignKey(User, on_delete=models.CASCADE)
+    original_file_name = models.TextField(max_length=2048, blank=True, null=True)
 
     @property
     def file_type(self):
         name = self.original_file.name
         return name.split(".")[-1]
 
     @property
-    def file_url(self):
+    def url(self):
         return self.original_file.url
 
     @property
     def name(self):
-        return self.original_file.name
+        return (
```

Comment:
the logic here is confusing me, what if `self.original_file_name` was `None` , `""` ?

Reply:
The names probably aren't helping tbf.

If django-storages sees a file with a duplicate name it will append random strings to the end. I'm storing the original name separately so it gets saved before django changes it in the filefield. This way people won't see `my_file_QWLSOX.pdf` in the list.

The check is to show the original name before showing the django-set file name, but with backwards compatibility for files that don't have it set yet in the db